realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -556,6 +571,8 @@ private void emitCodeForUnderConstruction(JavaWriter writer, boolean isPrimaryKe
                 .emitEmptyLine();
     }
 
+    // Note that because of bytecode hackery, this method may run before the constructor!
+    // It may ever run before fields have been initialized.
```

Comment:
That seems odd to me.  I have never seen a constructor see uninitialized fields.  I can guarantee that this method sometimes runs before fields are initialized.  I may be missing something but I don't see how it can run after the constructor, but before field initialization.  It is possible that it runs after superclass constructor.  Didn't check that.